src/standard/statement.ml

```diff
@@ -470,10 +473,10 @@ let p_inccnf ?loc () =
 let clause ?loc l = mk_clause ?loc l
 
 let assumption ?loc l =
-  mk ?loc (Prove l)
+  mk ?loc (Prove { hyps = l; goals = [] })
 
 (* Smtlib wrappers *)
-let check_sat ?loc l = mk ?loc (Prove l)
+let check_sat ?loc ?name hyps goals = mk ?loc ?id:name (Prove {hyps; goals})
```

Comment:
As said above: you never use the `goals` in this function (or rather you always provide the empty list), so there is no need for this argument. Additionally, since this is specifically a smtlib wrapper, I'm not particularly fond of adding a `?name` argument that would be unused in the smtlib parser, :/